posthog-server/src/main/java/com/posthog/server/internal/FlagDefinitionModels.kt

```diff
@@ -0,0 +1,86 @@
+package com.posthog.server.internal
+
+import com.google.gson.annotations.SerializedName
+
+/**
+ * Response from /api/feature_flag/local_evaluation/
+ */
+internal data class LocalEvaluationResponse(
+    val flags: List<FlagDefinition>?,
+    @SerializedName("group_type_mapping")
+    val groupTypeMapping: Map<String, String>?,
+    val cohorts: Map<String, CohortDefinition>?,
+)
+
+/**
+ * Complete feature flag definition for local evaluation
+ */
+internal data class FlagDefinition(
+    val id: Int,
+    val name: String,
+    val key: String,
+    val active: Boolean,
+    val filters: FlagFilters,
+    val version: Int,
+)
+
+/**
+ * Flag filters containing groups and multivariate config
+ */
+internal data class FlagFilters(
+    val groups: List<FlagConditionGroup>?,
+    val multivariate: MultiVariateConfig?,
+    val payloads: Map<String, Any?>?,
+)
+
+/**
+ * A condition group with properties and rollout percentage
+ */
+internal data class FlagConditionGroup(
+    val properties: List<FlagProperty>?,
+    @SerializedName("rollout_percentage")
+    val rolloutPercentage: Int?,
+    val variant: String?,
+)
+
+/**
+ * A property condition for flag evaluation
+ */
+internal data class FlagProperty(
+    val key: String,
+    val value: Any?,
+    val operator: String?,
```

Comment:
is `operator` a reserved name in kotlin or just gh diff coloring?
if its reserved, you can use backticks to escape the reserved name or rename and use `SerializedName`

Reply:
It is a Kotlin keyword, but in this case it'd be shadowed. I went back and forth on this a bit. Since you've called it out, I've changed these names.